include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -808,18 +807,15 @@ AffineConstraints<number>::close()
               ++entry;
         }
 
-      // if we didn't do anything in this round, then quit the loop
-      if (chained_constraint_replaced == false)
-        break;
-
 #ifdef DEBUG
       // increase iteration count. note that we should not iterate more
       // times than there are constraints, since this puts a natural upper
       // bound on the length of constraint chains
       ++iteration;
-      Assert(iteration <= lines.size(), ExcInternalError());
+      Assert(iteration <= lines.size() + 1, ExcInternalError());
```

Comment:
Since we had a `break` right above, we might get here one more time than before. So extend the check by one.

I don't think this assertion has ever triggered. So I'm not terribly concerned about the specific assertion's iteration count.